### ISSUE TITLE
dask dataframe drop_duplicates support?

### ISSUE THREAD

**Issue by apiszcz** [dask/dask#2735]
edit:

This works

```python
>>> df = pd.DataFrame({"A": [1, 2] * 4, "B": [1] * 4 + [2] * 4})
>>> a = dd.from_pandas(df, 2)
>>> a.drop_duplicates(subset=['A']).compute()
```

But this fails when `subset` is specified positionally.

```python
>>> a.drop_duplicates(['A']).compute()
```

```pytb
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-ad2309a8f442> in <module>
      3 df = dd.from_pandas(pd.DataFrame({'a': [0, 0, 1], 'b': range(3)}), npartitions=1)
      4
----> 5 df.drop_duplicates(['a']).compute()

~/sandbox/dask/dask/dataframe/core.py in drop_duplicates(self, split_every, split_out, **kwargs)
    506             split_out_setup=split_out_setup,
    507             split_out_setup_kwargs=split_out_setup_kwargs,
--> 508             **kwargs
    509         )
    510

~/sandbox/dask/dask/dataframe/core.py in apply_concat_apply(args, chunk, aggregate, combine, meta, token, chunk_kwargs, aggregate_kwargs, combine_kwargs, split_every, split_out, split_out_setup, split_out_setup_kwargs, **kwargs)
   4597     elif split_every is False:
   4598         split_every = npartitions
-> 4599     elif split_every < 2 or not isinstance(split_every, Integral):
   4600         raise ValueError("split_every must be an integer >= 2")
   4601

TypeError: '<' not supported between instances of 'list' and 'int'

```

We need to make `subset` a proper keyword in 

https://github.com/dask/dask/blob/96bbf636c50a753b76ba4758d03aebb0a42386db/dask/dataframe/core.py#L485

and update lines like 

https://github.com/dask/dask/blob/96bbf636c50a753b76ba4758d03aebb0a42386db/dask/dataframe/core.py#L488

to just be

```python
if subset is not None:
    ....
```

---

The original df is a dask dataframe

This works:
```
df = df.compute()
columns=['a','b','c']
df = df.drop_duplicates(columns)
```

This does not, is there a better approach?
```
columns=['a','b','c']
(Pdb) df = df.drop_duplicates(columns)
*** TypeError: unorderable types: list() < int()
```

**Comment by TomAugspurger** [comment]
Can you show a reproducible example? The following works for me:

```
In [19]: df = pd.DataFrame({"A": [1, 2] * 4, "B": [1] * 4 + [2] * 4})

In [20]: a = dd.from_pandas(df, 2)

In [21]: a.drop_duplicates(subset=['A']).compute()
Out[21]:
   A  B
0  1  1
1  2  1
```

**Comment by TomAugspurger** [comment]
Ah, keep in mind that the signature isn't identical. http://dask.pydata.org/en/latest/dataframe-api.html#dask.dataframe.DataFrame.drop_duplicates vs https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.drop_duplicates.html

We should probably deprecate dask's version to match pandas.

**Comment by mlisovyi** [comment]
The `subset` arg would be indeed useful (also documentation of the dask-specific args would be handy)

**Comment by daniel-severo** [comment]
Actually, I think this works fine and the docs are just outdated.

```python3
In [11]: import dask.dataframe as dd                
    ...: import pandas as pd
    ...: df = dd.from_pandas(pd.DataFrame({'a': [0, 0, 1], 'b': range(3)}), npartitions=1) 

In [12]: df.drop_duplicates(subset=['a']).compute() 
Out[12]:                     
   a  b                                                                                    
0  0  0                    
2  1  2 
```

**Comment by WesRoach** [comment]
It seems in @dsevero example drop_duplicates throws a non-relevant error when `subset=` isn't explicitly declared. @TomAugspurger is this intentional, or should we resolve with a PR?

```
import dask.dataframe as dd                
import pandas as pd
df = dd.from_pandas(pd.DataFrame({'a': [0, 0, 1], 'b': range(3)}), npartitions=1) 

df.drop_duplicates(['a']).compute() 
TypeError: '<' not supported between instances of 'list' and 'int'
```

**Comment by TomAugspurger** [comment]
Probably not intentional.

The fix is to add `subset=None` as the first keyword to `DataFrame.drop_duplicate`. That will fix the documentation issue as well.

Then update references like `subset in kwargs` inside drop_duplicates.

**Comment by TomAugspurger** [comment]
I updated the original post with hopefully clear instructions on fixing, in case anyone is interested in contributing a fix.

**Comment by WesRoach** [comment]
Interested. Reading contribution guidelines (previous contribution was docs).

**Comment by WesRoach** [comment]
I think this issue can be closed? I thought the `fixes` keyword in #5410 would close this issue on merge, but I guess not?

**Comment by TomAugspurger** [comment]
Yep, you're correct. Maybe it has to be in the first post.
